### ISSUE TITLE
Config defaults handling might boot to an unusable state

### ISSUE THREAD

**Issue by sdsnatcher** [openMSX/openMSX#1376]
The way openMSX handles some of the configs might cause it to boot to an unusable state, where it uses a lot of CPU and barely accepts any input.

Steps to reproduce:

1) Boot a Panasonic FS-A1WX
2) set the following configuration on the console:
set z80_freq 80000000
set too_fast_vram_access ignore
set z80_freq_locked off

3) Quit openMSX
4) Boot a Panasonic FS-A1WX again

openMSX will use a lot of CPU logging a lot of "too fast VRAM access" and barely accept any keyboard inputs. This makes it very difficult to open a console and try to restore a usable configuration.

Tested on:
- openMSX v17.0
- macOS 11.5.2

--------------------

The problem is caused by the different way openMSX deals with some configs related to the MSX speed on startup:

- z80_freq_locked and r800_freq_locked:  saved and restored between sessions
- too_fast_vram_access: not saved. Always booted with the default "real"
- cmdtiming: not saved. Always booted with the default "real"

IMHO, for the user/developer convenience, all the settings above should be saved/restored per-machine if possible.

**Comment by m9710797** [comment]
Thanks for your report. It's a valid issue, but I'm not sure I agree with the proposed solution.

My reasoning for auto-saving setting values (or not) was the following:
* Stuff that's purely debugging should not be saved (e.g. breakpoints are not saved).
* Stuff that also exists in real machines should be saved.

I understand that this classification does not cover everything, but for this specific ticket it's sufficient.

"too_fast_vram_access" and "cmdtiming" are clearly debug features, so these should not be saved. On the other hand, some people have modified their MSX machines to run at e.g. 7MHz, sometimes even with a physical switch to select between the 'original' and the 'overclocked' speed. The status of that switch does persist when shutting-down/restarting a real MSX machine. Similar to how some machines have a "firmware" switch and we save the status of that switch.

Another way to look at this:
* When using the "too_fast_vram_access" and/or "cmdtiming" feature, I'm probably a developer debugging some issue. Next time I start openMSX I probably want the normal behavior again. (Or as a developer I can easily write a custom startup script).
* Occasionally when playing a game that runs a bit sluggish on a 3.5MHz machine, I "overclock" the emulated machine. When continue playing the next day it was useful that my machine kept this setting.
* OTOH I've also encountered situations where I got confused about the saved "overclock" behavior.

I understand that this is tailored towards my personal use cases and that other people may have other use cases.

So I'm not sure what to do:
* Auto-saving "too_fast_vram_access" and "cmdtiming" seems the wrong approach to me.  It would introduce more problems than it solves.
* Not auto-saving "z80_freq_locked" seems like a better solution to me. But that's not consistent with e.g. saving the status of the "firmware-switch".
* All IMHO of course.

Any other opinions?

**Comment by mthuurne** [comment]
As far as I know, 7MHz conversions included some circuitry to add wait cycles after I/O to prevent too-fast access. So unless we emulate that as well, `z80_freq_locked` is not the exact equivalent of a 7MHz switch.

**Comment by MBilderbeek** [comment]
@mthuurne See #136 .... So, I tend to agree.

**Comment by sdsnatcher** [comment]
@mthuurne is right: I confirm that every turbo MSX has a way to prevent too-fast access to the VRAM and, depending on the CPU speed, the VDP registers too.

- Victor HC-90/HC-95: The HD64180 CPU has a software-configurable I/O waitstate generator. The amount of waitstates can be configured on its built-in BIOS SETUP utility.
- Panasonic MSX2+ machines: The turbo is implemented on the M60014-0140FP gate-array. It selects the CPU speed and generates the extra waitstates on the turbo mode.
- CIEL ExpertTurbo: The V9958 waitstate generator is used. This allows much faster VRAM transfers (almost 2x).
- Homebrew SuperTurbo and similar designs: They lower the clock to 3.57MHz for some time (~128 cycles) after every I/O instruction.

The z80_freq_locked and r800_freq_locked options are not usable to speed-up the emulated MSX if the too_fast_vram_access isn't enabled at the same time. It's more like a "what if" debug setting.

**Comment by m9710797** [comment]
I fully agree with you all: our `z80_freq_locked` setting is not fully the same as a 'real' 7MHz switch.
But it's less clear to me how this connects to the original question about saving or not-saving these settings.

**Comment by mthuurne** [comment]
I think it would make sense to not persist the value of `z80_freq_lock`.

**Comment by MBilderbeek** [comment]
Yes, we're trying to make the point that firmware switch and `z80_freq_locked` are not of the same category. So it's not a problem if they are not consistent with each other :)

**Comment by MBilderbeek** [comment]
@m9710797 So, let's make `z80_freq_locked` not persistent, agreed?

**Comment by sdsnatcher** [comment]
Was this already implemented on a daily build? I keep running on this problem with the latest stable release when I'm working on a turbofix.

**Comment by MBilderbeek** [comment]
This was handled via 2dd20a9411a2ab5e9157c7e5251119e60c0c10c2 and thus the duplicate #1729 issue.
